src/components/tables/AccessGrants/index.tsx

```diff
@@ -1,67 +1,60 @@
-import { SortDirection } from 'types';
-
-// const queryColumns = ['id', 'spec_type', 'catalog_name', 'updated_at'];
+import TableHydrator from 'stores/Tables/Hydrator';
+import useTableState from '../hooks';
 
 function AccessGrantsTable() {
-    const rowsPerPage = 10;
-    const [pagination, setPagination] = useState<{ from: number; to: number }>(
-        getPagination(0, rowsPerPage)
-    );
-    const [searchQuery, setSearchQuery] = useState<string | null>(null);
-    const [sortDirection, setSortDirection] = useState<SortDirection>('asc');
-    const [columnToSort, setColumnToSort] = useState<any>('user_full_name');
+    const {
+        pagination,
+        setPagination,
+        searchQuery,
+        setSearchQuery,
+        sortDirection,
+        setSortDirection,
+        columnToSort,
+        setColumnToSort,
+    } = useTableState('user_full_name');
 
-    const rolesQuery = useQuery<any>(
-        TABLES.COMBINED_GRANTS_EXT,
-        {
-            columns: `id, subject_role, object_role, capability, user_avatar_url, user_full_name, user_email, updated_at`,
-            count: 'exact',
-            filter: (query) => {
-                return defaultTableFilter<any>(
-                    query,
-                    ['user_full_name', 'subject_role', 'object_role'],
-                    searchQuery,
-                    [
-                        {
-                            col: columnToSort,
-                            direction: sortDirection,
-                        },
-                    ],
-                    pagination
-                );
+    const query = useMemo(() => {
+        return getGrantsForEverything(pagination, searchQuery, [
+            {
+                col: columnToSort,
+                direction: sortDirection,
             },
-        },
-        [pagination, searchQuery, columnToSort, sortDirection]
-    );
+        ]);
+    }, [columnToSort, pagination, searchQuery, sortDirection]);
 
     return (
         <Box>
-            <EntityTable
-                noExistingDataContentIds={{
-                    header: 'accessGrants.message1',
-                    message: 'accessGrants.message2',
-                    disableDoclink: true,
-                }}
-                columns={tableColumns}
-                query={rolesQuery}
-                renderTableRows={(data) => <Rows data={data} />}
-                setPagination={setPagination}
-                setSearchQuery={setSearchQuery}
-                sortDirection={sortDirection}
-                setSortDirection={setSortDirection}
-                columnToSort={columnToSort}
-                setColumnToSort={setColumnToSort}
-                header="accessGrantsTable.title"
-                filterLabel="accessGrantsTable.filterLabel"
+            <TableHydrator
+                query={query}
                 selectableTableStoreName={SelectTableStoreNames.ACCESS_GRANTS}
-            />
+            >
```

Comment:
The hydrator gets the query and is the one that actually puts all the stuff into the table store.